From 571a43a5257c74c016e5b18df0c9e79232cba7e4 Mon Sep 17 00:00:00 2001
From: Marius Volkhart <marius.volkhart@pkware.com>
Date: Thu, 25 Jun 2026 12:18:14 -0400
Subject: [PATCH] Replace grpc-netty-shaded with grpc-netty

Compile against real io.grpc:grpc-netty instead of the vendored shaded
copy so consuming applications control a single Netty version on their
classpath. Real grpc-netty exposes the same GrpcSslContexts/SslContext
APIs as the shaded variant, so this is a package-prefix de-shade
(io.grpc.netty.shaded.io.netty.* -> io.netty.*) rather than a rewrite;
the affected files revert to upstream form. Also de-shades the
temporal-envconfig sources and the native-image metadata in
temporal-serviceclient and temporal-test-server.
---
 .../envconfig/ClientConfigProfile.java        |   6 +-
 .../envconfig/ClientConfigProfileTest.java    |   2 +-
 ...GetVersionInterleavedUpdateReplayTest.java |   2 +-
 temporal-serviceclient/build.gradle           |   2 +-
 .../serviceclient/ChannelManager.java         |   2 +-
 .../serviceclient/ServiceStubsOptions.java    |   9 +-
 .../SimpleSslContextBuilder.java              |   8 +-
 .../native-image.properties                   |  12 +-
 .../reflect-config.json                       | 202 +++++++-------
 .../resource-config.json                      |   2 +-
 .../serviceclient/GrpcCompressionTest.java    |   2 +-
 .../reachability-metadata.json                | 250 +++++++++---------
 12 files changed, 249 insertions(+), 250 deletions(-)

diff --git a/temporal-envconfig/src/main/java/io/temporal/envconfig/ClientConfigProfile.java b/temporal-envconfig/src/main/java/io/temporal/envconfig/ClientConfigProfile.java
index 3257fbf..3db5b8f 100644
--- a/temporal-envconfig/src/main/java/io/temporal/envconfig/ClientConfigProfile.java
+++ b/temporal-envconfig/src/main/java/io/temporal/envconfig/ClientConfigProfile.java
@@ -1,9 +1,9 @@
 package io.temporal.envconfig;
 
 import io.grpc.Metadata;
-import io.grpc.netty.shaded.io.grpc.netty.GrpcSslContexts;
-import io.grpc.netty.shaded.io.netty.handler.ssl.SslContextBuilder;
-import io.grpc.netty.shaded.io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import io.grpc.netty.GrpcSslContexts;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
 import io.temporal.client.WorkflowClientOptions;
 import io.temporal.serviceclient.WorkflowServiceStubsOptions;
 import java.io.ByteArrayInputStream;
diff --git a/temporal-envconfig/src/test/java/io/temporal/envconfig/ClientConfigProfileTest.java b/temporal-envconfig/src/test/java/io/temporal/envconfig/ClientConfigProfileTest.java
index da4cc86..79572d2 100644
--- a/temporal-envconfig/src/test/java/io/temporal/envconfig/ClientConfigProfileTest.java
+++ b/temporal-envconfig/src/test/java/io/temporal/envconfig/ClientConfigProfileTest.java
@@ -3,7 +3,7 @@ package io.temporal.envconfig;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.io.Files;
 import io.grpc.Metadata;
-import io.grpc.netty.shaded.io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContext;
 import io.temporal.serviceclient.WorkflowServiceStubsOptions;
 import java.io.*;
 import java.nio.charset.StandardCharsets;
diff --git a/temporal-sdk/src/test/java/io/temporal/workflow/versionTests/GetVersionInterleavedUpdateReplayTest.java b/temporal-sdk/src/test/java/io/temporal/workflow/versionTests/GetVersionInterleavedUpdateReplayTest.java
index 5aa8813..ef86531 100644
--- a/temporal-sdk/src/test/java/io/temporal/workflow/versionTests/GetVersionInterleavedUpdateReplayTest.java
+++ b/temporal-sdk/src/test/java/io/temporal/workflow/versionTests/GetVersionInterleavedUpdateReplayTest.java
@@ -11,7 +11,7 @@ import io.grpc.MethodDescriptor;
 import io.grpc.Server;
 import io.grpc.ServerServiceDefinition;
 import io.grpc.Status;
-import io.grpc.netty.shaded.io.grpc.netty.NettyServerBuilder;
+import io.grpc.netty.NettyServerBuilder;
 import io.grpc.stub.ClientCalls;
 import io.grpc.stub.ServerCalls;
 import io.grpc.stub.StreamObserver;
diff --git a/temporal-serviceclient/build.gradle b/temporal-serviceclient/build.gradle
index e4b6976..215cbf2 100644
--- a/temporal-serviceclient/build.gradle
+++ b/temporal-serviceclient/build.gradle
@@ -11,7 +11,7 @@ dependencies {
 
     api ("io.grpc:grpc-api") //Classes like io.grpc.Metadata are used as a part of our API
     api "io.grpc:grpc-stub" //Part of WorkflowServiceStubs API
-    api "io.grpc:grpc-netty-shaded" //Part of WorkflowServiceStubs API, specifically SslContext
+    api "io.grpc:grpc-netty" //Part of WorkflowServiceStubs API, specifically SslContext
     api "io.grpc:grpc-protobuf" //For io.grpc.protobuf.StatusProto and ProtoUtils used by generated stubs
     api "io.grpc:grpc-services" //Standard gRPC HealthCheck Response class
     api "io.grpc:grpc-inprocess" //For the in-process time skipping test server
diff --git a/temporal-serviceclient/src/main/java/io/temporal/serviceclient/ChannelManager.java b/temporal-serviceclient/src/main/java/io/temporal/serviceclient/ChannelManager.java
index 4d32633..91472d3 100644
--- a/temporal-serviceclient/src/main/java/io/temporal/serviceclient/ChannelManager.java
+++ b/temporal-serviceclient/src/main/java/io/temporal/serviceclient/ChannelManager.java
@@ -5,7 +5,7 @@ import io.grpc.*;
 import io.grpc.health.v1.HealthCheckRequest;
 import io.grpc.health.v1.HealthCheckResponse;
 import io.grpc.health.v1.HealthGrpc;
-import io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder;
+import io.grpc.netty.NettyChannelBuilder;
 import io.grpc.stub.MetadataUtils;
 import io.temporal.api.workflowservice.v1.GetSystemInfoResponse.Capabilities;
 import io.temporal.internal.retryer.GrpcRetryer;
diff --git a/temporal-serviceclient/src/main/java/io/temporal/serviceclient/ServiceStubsOptions.java b/temporal-serviceclient/src/main/java/io/temporal/serviceclient/ServiceStubsOptions.java
index 4161dca..0789325 100644
--- a/temporal-serviceclient/src/main/java/io/temporal/serviceclient/ServiceStubsOptions.java
+++ b/temporal-serviceclient/src/main/java/io/temporal/serviceclient/ServiceStubsOptions.java
@@ -4,7 +4,7 @@ import com.google.common.base.MoreObjects;
 import com.uber.m3.tally.NoopScope;
 import com.uber.m3.tally.Scope;
 import io.grpc.*;
-import io.grpc.netty.shaded.io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContext;
 import io.temporal.authorization.AuthorizationGrpcMetadataProvider;
 import io.temporal.authorization.AuthorizationTokenSupplier;
 import java.time.Duration;
@@ -531,7 +531,7 @@ public class ServiceStubsOptions {
     /**
      * Gives an opportunity to provide some additional configuration to the channel builder or
      * override configurations done by the Temporal Stubs. Currently, Temporal Stubs use {@link
-     * io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder} to create a {@link ManagedChannel}.
+     * io.grpc.netty.NettyChannelBuilder} to create a {@link ManagedChannel}.
      *
      * <p>Advanced API
      *
@@ -539,9 +539,8 @@ public class ServiceStubsOptions {
      *
      * @param channelInitializer listener that will be called as a last step of channel creation if
      *     Stubs are configured with {@link Builder#setTarget(String)}. The listener is called with
-     *     an instance of {@link io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder} that will
-     *     be used by Temporal Stubs to create a {@link ManagedChannel}. The builder type may change
-     *     in the future.
+     *     an instance of {@link io.grpc.netty.NettyChannelBuilder} that will be used by Temporal
+     *     Stubs to create a {@link ManagedChannel}. The builder type may change in the future.
      * @return {@code this}
      */
     public T setChannelInitializer(Consumer<ManagedChannelBuilder<?>> channelInitializer) {
diff --git a/temporal-serviceclient/src/main/java/io/temporal/serviceclient/SimpleSslContextBuilder.java b/temporal-serviceclient/src/main/java/io/temporal/serviceclient/SimpleSslContextBuilder.java
index cabc70c..d14c471 100644
--- a/temporal-serviceclient/src/main/java/io/temporal/serviceclient/SimpleSslContextBuilder.java
+++ b/temporal-serviceclient/src/main/java/io/temporal/serviceclient/SimpleSslContextBuilder.java
@@ -1,9 +1,9 @@
 package io.temporal.serviceclient;
 
-import io.grpc.netty.shaded.io.grpc.netty.GrpcSslContexts;
-import io.grpc.netty.shaded.io.netty.handler.ssl.SslContext;
-import io.grpc.netty.shaded.io.netty.handler.ssl.SslContextBuilder;
-import io.grpc.netty.shaded.io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import io.grpc.netty.GrpcSslContexts;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
 import java.io.InputStream;
 import java.security.KeyStore;
 import java.security.KeyStoreException;
diff --git a/temporal-serviceclient/src/main/resources/META-INF/native-image/io/temporal/temporal-serviceclient/native-image.properties b/temporal-serviceclient/src/main/resources/META-INF/native-image/io/temporal/temporal-serviceclient/native-image.properties
index 4b61f30..888d59b 100644
--- a/temporal-serviceclient/src/main/resources/META-INF/native-image/io/temporal/temporal-serviceclient/native-image.properties
+++ b/temporal-serviceclient/src/main/resources/META-INF/native-image/io/temporal/temporal-serviceclient/native-image.properties
@@ -20,12 +20,12 @@
 
 Args = --initialize-at-build-time=org.slf4j.impl.StaticLoggerBinder \
   --initialize-at-build-time=org.slf4j.LoggerFactory \
-  --initialize-at-run-time=io.grpc.netty.shaded.io.netty.channel.epoll \
-  --initialize-at-run-time=io.grpc.netty.shaded.io.netty.channel.unix \
-  --initialize-at-run-time=io.grpc.netty.shaded.io.netty.util.internal.logging.Slf4JLoggerFactory$NopInstanceHolder \
-  --initialize-at-run-time=io.grpc.netty.shaded.io.netty.util.internal.logging.Log4JLogger \
-  --initialize-at-run-time=io.grpc.netty.shaded.io.netty.internal.tcnative \
-  --initialize-at-run-time=io.grpc.netty.shaded.io.netty.handler.ssl \
+  --initialize-at-run-time=io.netty.channel.epoll \
+  --initialize-at-run-time=io.netty.channel.unix \
+  --initialize-at-run-time=io.netty.util.internal.logging.Slf4JLoggerFactory$NopInstanceHolder \
+  --initialize-at-run-time=io.netty.util.internal.logging.Log4JLogger \
+  --initialize-at-run-time=io.netty.internal.tcnative \
+  --initialize-at-run-time=io.netty.handler.ssl \
   -H:DynamicProxyConfigurationResources=${.}/proxy-config.json \
   -H:JNIConfigurationResources=${.}/jni-config.json \
   -H:ReflectionConfigurationResources=${.}/reflect-config.json \
diff --git a/temporal-serviceclient/src/main/resources/META-INF/native-image/io/temporal/temporal-serviceclient/reflect-config.json b/temporal-serviceclient/src/main/resources/META-INF/native-image/io/temporal/temporal-serviceclient/reflect-config.json
index 44fa4dc..0ff016c 100644
--- a/temporal-serviceclient/src/main/resources/META-INF/native-image/io/temporal/temporal-serviceclient/reflect-config.json
+++ b/temporal-serviceclient/src/main/resources/META-INF/native-image/io/temporal/temporal-serviceclient/reflect-config.json
@@ -138,200 +138,200 @@
   "name":"io.grpc.internal.PickFirstLoadBalancerProvider"
 },
 {
-  "name":"io.grpc.netty.shaded.io.grpc.netty.AbstractNettyHandler",
+  "name":"io.grpc.netty.AbstractNettyHandler",
   "methods":[
-    {"name":"channelActive","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"exceptionCaught","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Throwable"] }
+    {"name":"channelActive","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"exceptionCaught","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Throwable"] }
   ]
 },
 {
-  "name":"io.grpc.netty.shaded.io.grpc.netty.NettyClientHandler",
+  "name":"io.grpc.netty.NettyClientHandler",
   "methods":[
-    {"name":"channelInactive","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"close","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"write","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Object","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] }
+    {"name":"channelInactive","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"close","parameterTypes":["io.netty.channel.ChannelHandlerContext","io.netty.channel.ChannelPromise"] },
+    {"name":"write","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Object","io.netty.channel.ChannelPromise"] }
   ]
 },
 {
-  "name":"io.grpc.netty.shaded.io.grpc.netty.ProtocolNegotiators$GrpcNegotiationHandler",
-  "methods":[{"name":"userEventTriggered","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Object"] }]
+  "name":"io.grpc.netty.ProtocolNegotiators$GrpcNegotiationHandler",
+  "methods":[{"name":"userEventTriggered","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Object"] }]
 },
 {
-  "name":"io.grpc.netty.shaded.io.grpc.netty.ProtocolNegotiators$ProtocolNegotiationHandler",
-  "methods":[{"name":"userEventTriggered","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Object"] }]
+  "name":"io.grpc.netty.ProtocolNegotiators$ProtocolNegotiationHandler",
+  "methods":[{"name":"userEventTriggered","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Object"] }]
 },
 {
-  "name":"io.grpc.netty.shaded.io.grpc.netty.ProtocolNegotiators$WaitUntilActiveHandler",
-  "methods":[{"name":"channelActive","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] }]
+  "name":"io.grpc.netty.ProtocolNegotiators$WaitUntilActiveHandler",
+  "methods":[{"name":"channelActive","parameterTypes":["io.netty.channel.ChannelHandlerContext"] }]
 },
 {
-  "name":"io.grpc.netty.shaded.io.grpc.netty.WriteBufferingAndExceptionHandler",
+  "name":"io.grpc.netty.WriteBufferingAndExceptionHandler",
   "methods":[
-    {"name":"channelInactive","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelRead","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Object"] },
-    {"name":"close","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"connect","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.net.SocketAddress","java.net.SocketAddress","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"exceptionCaught","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Throwable"] },
-    {"name":"flush","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"write","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Object","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] }
+    {"name":"channelInactive","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelRead","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Object"] },
+    {"name":"close","parameterTypes":["io.netty.channel.ChannelHandlerContext","io.netty.channel.ChannelPromise"] },
+    {"name":"connect","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.net.SocketAddress","java.net.SocketAddress","io.netty.channel.ChannelPromise"] },
+    {"name":"exceptionCaught","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Throwable"] },
+    {"name":"flush","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"write","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Object","io.netty.channel.ChannelPromise"] }
   ]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.buffer.AbstractByteBufAllocator",
+  "name":"io.netty.buffer.AbstractByteBufAllocator",
   "allDeclaredMethods":true
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.buffer.AbstractReferenceCountedByteBuf",
+  "name":"io.netty.buffer.AbstractReferenceCountedByteBuf",
   "fields":[{"name":"refCnt"}]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.channel.ChannelDuplexHandler",
+  "name":"io.netty.channel.ChannelDuplexHandler",
   "methods":[
-    {"name":"bind","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.net.SocketAddress","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"close","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"connect","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.net.SocketAddress","java.net.SocketAddress","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"deregister","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"disconnect","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"flush","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"read","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"write","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Object","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] }
+    {"name":"bind","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.net.SocketAddress","io.netty.channel.ChannelPromise"] },
+    {"name":"close","parameterTypes":["io.netty.channel.ChannelHandlerContext","io.netty.channel.ChannelPromise"] },
+    {"name":"connect","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.net.SocketAddress","java.net.SocketAddress","io.netty.channel.ChannelPromise"] },
+    {"name":"deregister","parameterTypes":["io.netty.channel.ChannelHandlerContext","io.netty.channel.ChannelPromise"] },
+    {"name":"disconnect","parameterTypes":["io.netty.channel.ChannelHandlerContext","io.netty.channel.ChannelPromise"] },
+    {"name":"flush","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"read","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"write","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Object","io.netty.channel.ChannelPromise"] }
   ]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.channel.ChannelInboundHandlerAdapter",
+  "name":"io.netty.channel.ChannelInboundHandlerAdapter",
   "methods":[
-    {"name":"channelActive","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelInactive","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelRead","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Object"] },
-    {"name":"channelReadComplete","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelRegistered","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelUnregistered","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelWritabilityChanged","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"exceptionCaught","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Throwable"] },
-    {"name":"userEventTriggered","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Object"] }
+    {"name":"channelActive","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelInactive","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelRead","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Object"] },
+    {"name":"channelReadComplete","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelRegistered","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelUnregistered","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelWritabilityChanged","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"exceptionCaught","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Throwable"] },
+    {"name":"userEventTriggered","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Object"] }
   ]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.channel.ChannelInitializer",
+  "name":"io.netty.channel.ChannelInitializer",
   "methods":[
-    {"name":"channelRegistered","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"exceptionCaught","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Throwable"] }
+    {"name":"channelRegistered","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"exceptionCaught","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Throwable"] }
   ]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.channel.DefaultChannelPipeline$HeadContext",
+  "name":"io.netty.channel.DefaultChannelPipeline$HeadContext",
   "methods":[
-    {"name":"bind","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.net.SocketAddress","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"channelActive","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelInactive","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelRead","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Object"] },
-    {"name":"channelReadComplete","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelRegistered","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelUnregistered","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelWritabilityChanged","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"close","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"connect","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.net.SocketAddress","java.net.SocketAddress","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"deregister","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"disconnect","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"exceptionCaught","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Throwable"] },
-    {"name":"flush","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"read","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"userEventTriggered","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Object"] },
-    {"name":"write","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Object","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] }
+    {"name":"bind","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.net.SocketAddress","io.netty.channel.ChannelPromise"] },
+    {"name":"channelActive","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelInactive","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelRead","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Object"] },
+    {"name":"channelReadComplete","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelRegistered","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelUnregistered","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelWritabilityChanged","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"close","parameterTypes":["io.netty.channel.ChannelHandlerContext","io.netty.channel.ChannelPromise"] },
+    {"name":"connect","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.net.SocketAddress","java.net.SocketAddress","io.netty.channel.ChannelPromise"] },
+    {"name":"deregister","parameterTypes":["io.netty.channel.ChannelHandlerContext","io.netty.channel.ChannelPromise"] },
+    {"name":"disconnect","parameterTypes":["io.netty.channel.ChannelHandlerContext","io.netty.channel.ChannelPromise"] },
+    {"name":"exceptionCaught","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Throwable"] },
+    {"name":"flush","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"read","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"userEventTriggered","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Object"] },
+    {"name":"write","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Object","io.netty.channel.ChannelPromise"] }
   ]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.channel.DefaultChannelPipeline$TailContext",
+  "name":"io.netty.channel.DefaultChannelPipeline$TailContext",
   "methods":[
-    {"name":"channelActive","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelInactive","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelRead","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Object"] },
-    {"name":"channelReadComplete","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelRegistered","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelUnregistered","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelWritabilityChanged","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"exceptionCaught","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Throwable"] },
-    {"name":"userEventTriggered","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Object"] }
+    {"name":"channelActive","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelInactive","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelRead","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Object"] },
+    {"name":"channelReadComplete","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelRegistered","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelUnregistered","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelWritabilityChanged","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"exceptionCaught","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Throwable"] },
+    {"name":"userEventTriggered","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Object"] }
   ]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.channel.DefaultFileRegion"
+  "name":"io.netty.channel.DefaultFileRegion"
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.channel.epoll.Epoll",
+  "name":"io.netty.channel.epoll.Epoll",
   "methods":[{"name":"isAvailable","parameterTypes":[] }]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.channel.epoll.EpollEventLoopGroup",
+  "name":"io.netty.channel.epoll.EpollEventLoopGroup",
   "methods":[{"name":"<init>","parameterTypes":["int","java.util.concurrent.ThreadFactory"] }]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.channel.epoll.NativeDatagramPacketArray$NativeDatagramPacket"
+  "name":"io.netty.channel.epoll.NativeDatagramPacketArray$NativeDatagramPacket"
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.channel.socket.nio.NioSocketChannel",
+  "name":"io.netty.channel.socket.nio.NioSocketChannel",
   "methods":[{"name":"<init>","parameterTypes":[] }]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.channel.unix.PeerCredentials"
+  "name":"io.netty.channel.unix.PeerCredentials"
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.channel.epoll.EpollSocketChannel"
+  "name":"io.netty.channel.epoll.EpollSocketChannel"
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.handler.codec.ByteToMessageDecoder",
+  "name":"io.netty.handler.codec.ByteToMessageDecoder",
   "methods":[
-    {"name":"channelRead","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Object"] },
-    {"name":"userEventTriggered","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.lang.Object"] }
+    {"name":"channelRead","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Object"] },
+    {"name":"userEventTriggered","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.lang.Object"] }
   ]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.handler.codec.http2.Http2ConnectionHandler",
+  "name":"io.netty.handler.codec.http2.Http2ConnectionHandler",
   "methods":[
-    {"name":"bind","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.net.SocketAddress","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"channelReadComplete","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"channelWritabilityChanged","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"connect","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","java.net.SocketAddress","java.net.SocketAddress","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"deregister","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"disconnect","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext","io.grpc.netty.shaded.io.netty.channel.ChannelPromise"] },
-    {"name":"flush","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] },
-    {"name":"read","parameterTypes":["io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"] }
+    {"name":"bind","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.net.SocketAddress","io.netty.channel.ChannelPromise"] },
+    {"name":"channelReadComplete","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"channelWritabilityChanged","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"connect","parameterTypes":["io.netty.channel.ChannelHandlerContext","java.net.SocketAddress","java.net.SocketAddress","io.netty.channel.ChannelPromise"] },
+    {"name":"deregister","parameterTypes":["io.netty.channel.ChannelHandlerContext","io.netty.channel.ChannelPromise"] },
+    {"name":"disconnect","parameterTypes":["io.netty.channel.ChannelHandlerContext","io.netty.channel.ChannelPromise"] },
+    {"name":"flush","parameterTypes":["io.netty.channel.ChannelHandlerContext"] },
+    {"name":"read","parameterTypes":["io.netty.channel.ChannelHandlerContext"] }
   ]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.util.AbstractReferenceCounted",
+  "name":"io.netty.util.AbstractReferenceCounted",
   "fields":[{"name":"refCnt"}]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.util.ReferenceCountUtil",
+  "name":"io.netty.util.ReferenceCountUtil",
   "allDeclaredMethods":true
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.util.internal.NativeLibraryUtil",
+  "name":"io.netty.util.internal.NativeLibraryUtil",
   "methods":[{"name":"loadLibrary","parameterTypes":["java.lang.String","boolean"] }]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.util.internal.shaded.org.jctools.queues.BaseMpscLinkedArrayQueueColdProducerFields",
+  "name":"io.netty.util.internal.shaded.org.jctools.queues.BaseMpscLinkedArrayQueueColdProducerFields",
   "fields":[{"name":"producerLimit"}]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.util.internal.shaded.org.jctools.queues.BaseMpscLinkedArrayQueueConsumerFields",
+  "name":"io.netty.util.internal.shaded.org.jctools.queues.BaseMpscLinkedArrayQueueConsumerFields",
   "fields":[{"name":"consumerIndex"}]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.util.internal.shaded.org.jctools.queues.BaseMpscLinkedArrayQueueProducerFields",
+  "name":"io.netty.util.internal.shaded.org.jctools.queues.BaseMpscLinkedArrayQueueProducerFields",
   "fields":[{"name":"producerIndex"}]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueConsumerIndexField",
+  "name":"io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueConsumerIndexField",
   "fields":[{"name":"consumerIndex"}]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueProducerIndexField",
+  "name":"io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueProducerIndexField",
   "fields":[{"name":"producerIndex"}]
 },
 {
-  "name":"io.grpc.netty.shaded.io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueProducerLimitField",
+  "name":"io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueProducerLimitField",
   "fields":[{"name":"producerLimit"}]
 },
 {
diff --git a/temporal-serviceclient/src/main/resources/META-INF/native-image/io/temporal/temporal-serviceclient/resource-config.json b/temporal-serviceclient/src/main/resources/META-INF/native-image/io/temporal/temporal-serviceclient/resource-config.json
index a260699..b6c925c 100644
--- a/temporal-serviceclient/src/main/resources/META-INF/native-image/io/temporal/temporal-serviceclient/resource-config.json
+++ b/temporal-serviceclient/src/main/resources/META-INF/native-image/io/temporal/temporal-serviceclient/resource-config.json
@@ -1,7 +1,7 @@
 {
   "resources":{
   "includes":[
-    {"pattern":"\\QMETA-INF/native/libio_grpc_netty_shaded_netty_transport_native_epoll_x86_64.so\\E"},
+    {"pattern":"\\QMETA-INF/native/libnetty_transport_native_epoll_x86_64.so\\E"},
     {"pattern":"\\QMETA-INF/services/io.grpc.LoadBalancerProvider\\E"},
     {"pattern":"\\QMETA-INF/services/io.grpc.NameResolverProvider\\E"},
     {"pattern":"\\QMETA-INF/services/org.slf4j.spi.SLF4JServiceProvider\\E"},
diff --git a/temporal-serviceclient/src/test/java/io/temporal/serviceclient/GrpcCompressionTest.java b/temporal-serviceclient/src/test/java/io/temporal/serviceclient/GrpcCompressionTest.java
index 12afed8..13b39fd 100644
--- a/temporal-serviceclient/src/test/java/io/temporal/serviceclient/GrpcCompressionTest.java
+++ b/temporal-serviceclient/src/test/java/io/temporal/serviceclient/GrpcCompressionTest.java
@@ -12,7 +12,7 @@ import io.grpc.ServerInterceptor;
 import io.grpc.ServerInterceptors;
 import io.grpc.Status;
 import io.grpc.StatusRuntimeException;
-import io.grpc.netty.shaded.io.grpc.netty.NettyServerBuilder;
+import io.grpc.netty.NettyServerBuilder;
 import io.grpc.stub.StreamObserver;
 import io.grpc.testing.GrpcCleanupRule;
 import io.temporal.api.workflowservice.v1.GetSystemInfoRequest;
diff --git a/temporal-test-server/src/main/resources/META-INF/native-image/io.temporal/temporal-test-server/reachability-metadata.json b/temporal-test-server/src/main/resources/META-INF/native-image/io.temporal/temporal-test-server/reachability-metadata.json
index 5a618cf..6105ae1 100644
--- a/temporal-test-server/src/main/resources/META-INF/native-image/io.temporal/temporal-test-server/reachability-metadata.json
+++ b/temporal-test-server/src/main/resources/META-INF/native-image/io.temporal/temporal-test-server/reachability-metadata.json
@@ -285,172 +285,172 @@
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.grpc.netty.AbstractNettyHandler",
+      "type": "io.grpc.netty.AbstractNettyHandler",
       "methods": [
         {
           "name": "channelActive",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "exceptionCaught",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Throwable"
           ]
         }
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.grpc.netty.NettyServer$1"
+      "type": "io.grpc.netty.NettyServer$1"
     },
     {
-      "type": "io.grpc.netty.shaded.io.grpc.netty.NettyServerHandler",
+      "type": "io.grpc.netty.NettyServerHandler",
       "methods": [
         {
           "name": "channelInactive",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "close",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "write",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelPromise"
           ]
         }
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.grpc.netty.ProtocolNegotiators$GrpcNegotiationHandler",
+      "type": "io.grpc.netty.ProtocolNegotiators$GrpcNegotiationHandler",
       "methods": [
         {
           "name": "userEventTriggered",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object"
           ]
         }
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.grpc.netty.ProtocolNegotiators$ProtocolNegotiationHandler",
+      "type": "io.grpc.netty.ProtocolNegotiators$ProtocolNegotiationHandler",
       "methods": [
         {
           "name": "userEventTriggered",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object"
           ]
         }
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.grpc.netty.ProtocolNegotiators$WaitUntilActiveHandler",
+      "type": "io.grpc.netty.ProtocolNegotiators$WaitUntilActiveHandler",
       "methods": [
         {
           "name": "channelActive",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         }
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.grpc.netty.UdsNameResolverProvider"
+      "type": "io.grpc.netty.UdsNameResolverProvider"
     },
     {
-      "type": "io.grpc.netty.shaded.io.grpc.netty.WriteBufferingAndExceptionHandler",
+      "type": "io.grpc.netty.WriteBufferingAndExceptionHandler",
       "methods": [
         {
           "name": "channelInactive",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelRead",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object"
           ]
         },
         {
           "name": "close",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "connect",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.net.SocketAddress",
             "java.net.SocketAddress",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "exceptionCaught",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Throwable"
           ]
         },
         {
           "name": "flush",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "write",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelPromise"
           ]
         }
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.bootstrap.ServerBootstrap$1"
+      "type": "io.netty.bootstrap.ServerBootstrap$1"
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.bootstrap.ServerBootstrap$ServerBootstrapAcceptor",
+      "type": "io.netty.bootstrap.ServerBootstrap$ServerBootstrapAcceptor",
       "methods": [
         {
           "name": "channelRead",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object"
           ]
         },
         {
           "name": "exceptionCaught",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Throwable"
           ]
         }
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.buffer.AbstractByteBufAllocator"
+      "type": "io.netty.buffer.AbstractByteBufAllocator"
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.buffer.AbstractReferenceCountedByteBuf",
+      "type": "io.netty.buffer.AbstractReferenceCountedByteBuf",
       "fields": [
         {
           "name": "refCnt"
@@ -458,335 +458,335 @@
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.channel.ChannelDuplexHandler",
+      "type": "io.netty.channel.ChannelDuplexHandler",
       "methods": [
         {
           "name": "bind",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.net.SocketAddress",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "close",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "connect",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.net.SocketAddress",
             "java.net.SocketAddress",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "deregister",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "disconnect",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "flush",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "read",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "write",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelPromise"
           ]
         }
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.channel.ChannelInboundHandlerAdapter",
+      "type": "io.netty.channel.ChannelInboundHandlerAdapter",
       "methods": [
         {
           "name": "channelActive",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelInactive",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelRead",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object"
           ]
         },
         {
           "name": "channelReadComplete",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelRegistered",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelUnregistered",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelWritabilityChanged",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "exceptionCaught",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Throwable"
           ]
         },
         {
           "name": "userEventTriggered",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object"
           ]
         }
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.channel.ChannelInitializer",
+      "type": "io.netty.channel.ChannelInitializer",
       "methods": [
         {
           "name": "channelRegistered",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "exceptionCaught",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Throwable"
           ]
         }
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.channel.DefaultChannelPipeline$HeadContext",
+      "type": "io.netty.channel.DefaultChannelPipeline$HeadContext",
       "methods": [
         {
           "name": "bind",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.net.SocketAddress",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "channelActive",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelInactive",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelRead",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object"
           ]
         },
         {
           "name": "channelReadComplete",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelRegistered",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelUnregistered",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelWritabilityChanged",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "close",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "connect",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.net.SocketAddress",
             "java.net.SocketAddress",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "deregister",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "disconnect",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "exceptionCaught",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Throwable"
           ]
         },
         {
           "name": "flush",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "read",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "userEventTriggered",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object"
           ]
         },
         {
           "name": "write",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelPromise"
           ]
         }
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.channel.DefaultChannelPipeline$TailContext",
+      "type": "io.netty.channel.DefaultChannelPipeline$TailContext",
       "methods": [
         {
           "name": "channelActive",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelInactive",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelRead",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object"
           ]
         },
         {
           "name": "channelReadComplete",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelRegistered",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelUnregistered",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelWritabilityChanged",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "exceptionCaught",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Throwable"
           ]
         },
         {
           "name": "userEventTriggered",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object"
           ]
         }
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.channel.DefaultFileRegion"
+      "type": "io.netty.channel.DefaultFileRegion"
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.channel.epoll.Epoll",
+      "type": "io.netty.channel.epoll.Epoll",
       "methods": [
         {
           "name": "isAvailable",
@@ -799,7 +799,7 @@
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.channel.epoll.EpollServerSocketChannel",
+      "type": "io.netty.channel.epoll.EpollServerSocketChannel",
       "methods": [
         {
           "name": "<init>",
@@ -808,92 +808,92 @@
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.channel.epoll.NativeDatagramPacketArray$NativeDatagramPacket"
+      "type": "io.netty.channel.epoll.NativeDatagramPacketArray$NativeDatagramPacket"
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.channel.unix.PeerCredentials"
+      "type": "io.netty.channel.unix.PeerCredentials"
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.handler.codec.ByteToMessageDecoder",
+      "type": "io.netty.handler.codec.ByteToMessageDecoder",
       "methods": [
         {
           "name": "channelRead",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object"
           ]
         },
         {
           "name": "userEventTriggered",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.lang.Object"
           ]
         }
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.handler.codec.http2.Http2ConnectionHandler",
+      "type": "io.netty.handler.codec.http2.Http2ConnectionHandler",
       "methods": [
         {
           "name": "bind",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.net.SocketAddress",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "channelReadComplete",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "channelWritabilityChanged",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "connect",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelHandlerContext",
             "java.net.SocketAddress",
             "java.net.SocketAddress",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "deregister",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "disconnect",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext",
-            "io.grpc.netty.shaded.io.netty.channel.ChannelPromise"
+            "io.netty.channel.ChannelHandlerContext",
+            "io.netty.channel.ChannelPromise"
           ]
         },
         {
           "name": "flush",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         },
         {
           "name": "read",
           "parameterTypes": [
-            "io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext"
+            "io.netty.channel.ChannelHandlerContext"
           ]
         }
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.util.AbstractReferenceCounted",
+      "type": "io.netty.util.AbstractReferenceCounted",
       "fields": [
         {
           "name": "refCnt"
@@ -901,10 +901,10 @@
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.util.ReferenceCountUtil"
+      "type": "io.netty.util.ReferenceCountUtil"
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.util.internal.shaded.org.jctools.queues.BaseMpscLinkedArrayQueueColdProducerFields",
+      "type": "io.netty.util.internal.shaded.org.jctools.queues.BaseMpscLinkedArrayQueueColdProducerFields",
       "fields": [
         {
           "name": "producerLimit"
@@ -912,7 +912,7 @@
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.util.internal.shaded.org.jctools.queues.BaseMpscLinkedArrayQueueConsumerFields",
+      "type": "io.netty.util.internal.shaded.org.jctools.queues.BaseMpscLinkedArrayQueueConsumerFields",
       "fields": [
         {
           "name": "consumerIndex"
@@ -920,7 +920,7 @@
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.util.internal.shaded.org.jctools.queues.BaseMpscLinkedArrayQueueProducerFields",
+      "type": "io.netty.util.internal.shaded.org.jctools.queues.BaseMpscLinkedArrayQueueProducerFields",
       "fields": [
         {
           "name": "producerIndex"
@@ -928,7 +928,7 @@
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueConsumerIndexField",
+      "type": "io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueConsumerIndexField",
       "fields": [
         {
           "name": "consumerIndex"
@@ -936,7 +936,7 @@
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueProducerIndexField",
+      "type": "io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueProducerIndexField",
       "fields": [
         {
           "name": "producerIndex"
@@ -944,7 +944,7 @@
       ]
     },
     {
-      "type": "io.grpc.netty.shaded.io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueProducerLimitField",
+      "type": "io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueProducerLimitField",
       "fields": [
         {
           "name": "producerLimit"
-- 
2.50.1 (Apple Git-155)

